src/apprt/gtk/Window.zig

```diff
@@ -883,11 +880,7 @@ fn gtkActionClose(
     ud: ?*anyopaque,
 ) callconv(.C) void {
     const self: *Window = @ptrCast(@alignCast(ud orelse return));
-    const surface = self.actionSurface() orelse return;
-    _ = surface.performBindingAction(.{ .close_surface = {} }) catch |err| {
```

Comment:
I'm speculating as to why this is here, but if you're also running into the issue where the window is closed and freed twice, then I would probably suggest to remove the logic for closing the window inside the tab view instead

Reply:
It's because `gtkActionClose` is associated with the "Close Window" menu item, not "Close a Surface". Nothing about windows being close/freed twice.